src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java

```diff
@@ -514,6 +520,39 @@ public Terminal getTerminalFromIdentifiable(Network network, String equipmentId,
         return null;
     }
 
+    public Set<Terminal> getTerminalsFromIdentifiable(@NonNull Identifiable<?> identifiable) {
+        if (identifiable instanceof Branch) {
+            Branch<?> branch = (Branch<?>) identifiable;
+            return Stream.of(
+                    branch.getTerminal1(),
+                    branch.getTerminal2()
+            ).collect(Collectors.toSet());
+        } else if (identifiable instanceof ThreeWindingsTransformer) {
+            ThreeWindingsTransformer w3t = (ThreeWindingsTransformer) identifiable;
+            return Stream.of(
+                    w3t.getLeg1().getTerminal(),
+                    w3t.getLeg2().getTerminal(),
+                    w3t.getLeg3().getTerminal()
+            ).collect(Collectors.toSet());
+        } else if (identifiable instanceof HvdcLine) {
+            HvdcLine hvdcLine = (HvdcLine) identifiable;
+            return Stream.of(
+                    hvdcLine.getConverterStation1().getTerminal(),
+                    hvdcLine.getConverterStation2().getTerminal()
+            ).collect(Collectors.toSet());
+        }
+        throw NetworkModificationException.createEquipmentTypeUnknown(identifiable.getClass().getSimpleName());
```

Comment:
This methos is specific to the operation status modification
Move it
And throw `equipmentTypeNotSupported`
Use powsybl `getTerminals` method